mudslide/models.py

```diff
@@ -553,12 +562,205 @@ def dV(self, X: ArrayLike) -> ArrayLike:
         return out.reshape([5, 2, 2])
 
 
+class TMModel(AdiabaticModel_):
+    def __init__(
+        self,
+        turbomole_dir: str,
+        states: ArrayLike,
+        sub_dir_stem:str = "traj", 
+        representation: str = "adiabatic",
+        reference: Any = None,
+    ):
+        AdiabaticModel_.__init__(self, representation=representation, reference=reference)
+
+        self.turbomole_dir = turbomole_dir
+        self.states = states
+        self.nstates_ = len(self.states)
+        self.sub_dir_stem = sub_dir_stem
+        self.sub_dir_num = 0
+
+        self.turbomole_init()
+
+    def turbomole_init(self):
+        self.coord_path = Path(self.turbomole_dir)/"control"  
+        self.get_coords()
+        self.call_and_parse_turbomole()
```

Comment:
Why does the initialization calculate the first step? That's not in keeping with how the rest of the models are designed and I don't think it's a good design.

Reply:
resolved!